File: app/src/main/java/com/qwe7002/root_kit/radio.java
package com.qwe7002.root_kit;


public class radio {
    public static boolean is_NR_connected() {
        return shell.run_shell_command("dumpsys telephony.registry |grep mServiceState|grep nrState=CONNECTED");
    }

    public static boolean is_NR_standby() {
        return shell.run_shell_command("dumpsys telephony.registry |grep mServiceState|grep -E \"nrState=NOT_RESTRICTED|nrState=RESTRICTED\"");
    }

    public static boolean is_LTE_CA() {
        return shell.run_shell_command("dumpsys telephony.registry |grep mServiceState|grep \"mIsUsingCarrierAggregation = true\"");
    }
}

File: app/src/main/java/com/qwe7002/telegram_rc/root_kit/Radio.kt
package com.qwe7002.telegram_rc.root_kit


object Radio {
    @JvmStatic
    val isNRConnected: Boolean
        get() = Shell.runShellCommand("dumpsys telephony.registry |grep mServiceState|grep nrState=CONNECTED")

    @JvmStatic
    val isNRStandby: Boolean
        get() = Shell.runShellCommand("dumpsys telephony.registry |grep mServiceState|grep -E \"nrState=NOT_RESTRICTED|nrState=RESTRICTED\"")

    @JvmStatic
    val isLTECA: Boolean
        get() = Shell.runShellCommand("dumpsys telephony.registry |grep mServiceState|grep \"mIsUsingCarrierAggregation = true\"")
}
